mitodl/ocw-hugo-themes
# set up e2e tests with playwright

#### Pre-Flight checklist


#### What are the relevant tickets?
#919 

#### What's this PR do?
This PR adds e2e testing via Playwright to OCW Hugo Themes.

There are a variety of other significant changes that I will try to call out in individual comments.

#### How should this be manually tested?

**First:** Run `yarn install` and read the readme:  https://github.com/mitodl/ocw-hugo-themes/tree/cc/ci2/tests-e2e  If you use VS Code, I highly recommend installing the Playwright extension. (See the Readme.).

1. Run `yarn test:e2e`. It should fail. Add the required environment variable. (Use value 3001; see https://github.com/mitodl/ocw-hugo-themes/pull/941#discussion_r1010893053 ).
2. Run `yarn test:e2e` again. It should suceed but it will also be kinda slow.
3. Run `yarn start:webpack`. In a separate shell, run `yarn test:e2e`. It should be pretty fast. (Now Playwright is reusing the webpack server.)
4. Stick a `.only` on one of the tests, and run `yarn test:e2e --debug`. You can step through the test. 
5. Run `yarn start:course` and `yarn start:www`, too. I changed some stuff with environment variables and scripts. Those should both still work. You could also try `OCW_TEST_COURSE=different-from-value-in-env-file yarn start:course`. That should work now, whereas previously it would not. 

#### Where should the reviewer start?
I suggest starting with [the readme](https://github.com/mitodl/ocw-hugo-themes/tree/cc/ci2/tests-e2e), **then** reading some of the test files. The test files should more or less make sense without context.

Then look at all the other junk.
